tools/mo/openvino/tools/mo/ops/topk.py

```diff
@@ -50,8 +51,7 @@ def infer(node):
 
         k = node.in_port(1).data.get_value()
         if k is None:
-            raise Error('The value defining number of output elements for layer "{}" is not defined'
-                        ''.format(node.soft_get('name')))
+            k = dynamic_dimension
```

Comment:
It doesn't look to be specified in the spec, but according to https://github.com/openvinotoolkit/openvino/blob/master/src/core/shape_inference/include/topk_shape_inference.hpp#L78, the dimension `output_shape[node.axis]` which we are going to set to this `k` value later in the code may also be dynamic even with static/constant `k`. I mean that this code still doesn't give the most precise shape inference. I hope it doesn't matter because the main purpose of the current PR is to avoid MO crashing, but the more accurate shape inference is what C++ operational part will cover. Is my understanding correct?

Reply:
I provided a link to the shape inference code for TopK operation where it can be less than `k`. The spec is not accurate on that thing. My concern is not about your code but about the remining part which always set static dimension in the output regardless of input.

Reply:
Ok, I got it. The current fix is correct and your (corner) case is not affected with my fix. Now we need to create a ticket to explore behavior of the current version of *TopK* and assess how it is aligned with FWs